### DISCUSSION TITLE
v3.1.x

### DISCUSSION THREAD

**Discussion by bjlittle** [SciTools/iris#4072] [Releases]
The release manager for `v3.1.x` is @pp-mo

The release feature branch is [v3.1.x](https://github.com/SciTools/iris/tree/v3.1.x)

| Minor Release | Status | Milestone | Project | Date | RTD | Release | conda-forge | PyPI |
| :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| `v3.1a0`† | Completed | [v3.1.0](https://github.com/SciTools/iris/milestone/42) | [Iris v3.1.0](https://github.com/SciTools/iris/projects/26) | 03/05/2021 | - | - | - | - |
| `v3.1.0rc0` | Completed | [v3.1.0](https://github.com/SciTools/iris/milestone/42) | [Iris v3.1.0](https://github.com/SciTools/iris/projects/26) | 17/08/2021 | [Whatsnew](https://scitools-iris.readthedocs.io/en/v3.1.0rc0/whatsnew/3.1.html) | [v3.1.0rc0](https://github.com/SciTools/iris/releases/tag/v3.1.0rc0) | [Available](https://anaconda.org/conda-forge/iris?version=3.1.0rc0&channel=rc_iris) | - |
| `v3.1.0` | Completed | [v3.1.0](https://github.com/SciTools/iris/milestone/42) | [Iris v3.1.0](https://github.com/SciTools/iris/projects/26) | 17/09/2021 | [Whatsnew](https://scitools-iris.readthedocs.io/en/v3.1.0/whatsnew/3.1.html) | [v3.1.0](https://github.com/SciTools/iris/releases/tag/v3.1.0) | [Available](https://anaconda.org/conda-forge/iris?version=3.1.0&channel=main) | [Available](https://pypi.org/project/scitools-iris/3.1.0/) |

Notes:
* † - `v3.1a0` is an internal engineering tag.

**Comment by pp-mo** [comment]
Final planning for content now happening (at last) !

### Overall plan:
 * we are still intending to get an **RC out for the end of August 2021** : working on this now
 * we simply don't have time now to include the prototype unstructured(UGRID) support in the [mesh-data-model feature-branch]().
   * which is a shame as this could be disruptive and we want to get it "out there", but for now it must wait.
   * .. but equally, mesh-data-model is still not quite ready, and this simplifies the 3.1 task massively

I will shortly be adding notes here on proposed content, including : 
 * tasks; 
 * outstanding PRs; 
 * whatsnew additions and missing content

**Comment by pp-mo** [comment]
## PRs to be included
The list should now be complete on the [3.1.0 release project](https://github.com/SciTools/iris/projects/26)
However, not everything there is 100% ready to go, and we might still have to "pull" some of them from the release.

These are all the possibles + their status, from the project columns :
**Waiting for review**
  * https://github.com/SciTools/iris/pull/3710
    * possibly all-done.  **Pinged contributor**.  Awaiting response + review signoff/merge from dev
  * https://github.com/SciTools/iris/pull/4206
    * Needs reviewing.  Also possibly a whatsnew entry (**@pp-mo working on that**)

**In progress**
  * https://github.com/SciTools/iris/pull/3882
    * not quite ready.  **Pinged contributor**.  Not vital, but we will complete this + include it if we can
  * https://github.com/SciTools/iris/pull/3790
    * status slightly unclear, may in fact be ready for review.  **Pinged contributor**
  * https://github.com/SciTools/iris/pull/4052
    * needing at least rebase + review. **Pinged contributor**
  * https://github.com/SciTools/iris/pull/4176
    * still in draft, and rather wanted by ESMValTool people.  
    * this one is on **me (@pp-mo)** : [status: will fix in ~2weeks when back from holiday, or _someone else could take it on ?_](https://github.com/SciTools/iris/pull/4176#issuecomment-889769057)

 **Backlog**
There are the 3 outstanding tasks relating to improvements/fixes to the notebook Cube formatting `Cube._repr_html()`
All from @DPeterK , who I have now **pinged**
  * #3376 
  * #3380 
  * #3383
  * **NOTE:** I moved **_all_** these to 'TODO', below the MVP-divider, for two reasons: 
    1. we suspect there may not be time to get these in (they have been stale for a while)
    2. they may be affected by #4206, or at least better re-built around #3987

**Reply by rcomer** [reply]
I just added #4263 to the project, as it is a fix for a problem I introduced since v3.0 😬

**Comment by pp-mo** [comment]
## Notes on additional Whatsnew content
I checked out all the commits from v3.0.0 to current 3.0.x head, looking for any significant changes or contributions lacking a whatsnew entry.
Which results in this list ...
  * #4115  : worth a mention, in the "FIXES" section?
  * #3941 : in "Internal"
  * #4124  : this **_does_** appear in v3.0.4, but quoting two different PRs against 3.0.x : #4233 and #4238
  * #4198 : should really have an "Internal" section entry
    * propose text something like "removed the dependency on Pyke for netcdf loading, replaced with pure Python."
  * #3987 : should be mentioned in entry for #4206, assuming that gets in
  * #4052 missing whatsnew

There have also been a **_lot_** of small improvements to documentation content + presentation, mostly from @tkknight and @bjlittle .  It might be worth saying _something_ about this, but I'm strugggling to summarise the changes.

All best dealt with in final stages of release definition, I think.

**Reply by rcomer** [reply]
For #3941, I figured it was so minor it didn’t warrant a whatsnew entry.  If you disagree, it could just be added to point 1 under documentation (other gallery updates).

https://github.com/SciTools/iris/blob/eca4fa16ece4e1a001420f939a4b6062824a1981/docs/src/whatsnew/latest.rst#-documentation

**Reply by pp-mo** [reply]
Thanks @rcomer .  You're right -- no-one is still waiting to be told that we don't support mpl 1.x !

**Comment by pp-mo** [comment]
I haven't seen any suggestions to modify the 3.1 release.
However, from discussion I am thinking that a couple of things could do with a bit more prominence in the whats-new page.
Specifically:
  * minimal sub-package imports are more consistent, ***but will break user code***
    * e.g. `import iris; cube = iris.cube.Cube(0)`
  * speedup for netcdf single-cube loading #4176

**Comment by pp-mo** [comment]
FYI I have now cut the v3.1.0 release : https://github.com/SciTools/iris/releases/tag/v3.1.0

The docs build is up already, and we should shortly have this on conda-forge and PyPI.

Thanks to all involved !

**Comment by pp-mo** [comment]
3.1.0 is now finalised + up on conda-forge and PyPI
mergeback to 'main' has also been done https://github.com/SciTools/iris/pull/4325

**Reply by rcomer** [reply]
Nice work @pp-mo 🎉
